tests/benchmarks/context_benchmark.py

```diff
@@ -0,0 +1,122 @@
+"""
+Given a codebase, a prompt and a token limit, code_context will auto-select
+features to include in the code_message. This script evaluates the 
+performance of that selection algorithm.
+
+Test Guidelines:
+- Codebase should be larger than the token limit
+- Prompts should be thorough and specific - give it the best chance to succeed
+- Tasks should draw on multiple files/features.
+
+Scoring:
+- Score based on the features (path, level, diff) auto-selected
+- Use traditional precision, recall and f1 scores:
+    - y_pred = [(f in code_context.features) for f in all_features]
+    - y_true = [(f in test.expected) for f in all_features]
+
+Fine-tuning:
+An auto-context process might look like this:
+1. Use local methods (embeddings) to select a full-context worth
+    of features
+2. Send the code_message of those features to gpt-3, and have it 
+    return a modified features spec [(path, level, diff)] based on what's
+    relevant and not. (Bonus: let gpt-3 ask for more info if it needs it)
+3. Use the modified features to get_code_message for gpt-4
+
+These tests can be used to train/score #1 or #2, but we'd expect #2 to score 
+a lot higher.
+"""
+import os
+
+import pytest
+
+from mentat.code_context import CodeContext, CodeContextSettings
+from mentat.code_file import CodeFile, CodeMessageLevel
+from mentat.git_handler import GIT_ROOT, get_non_gitignored_files
+from mentat.llm_api import setup_api_key
+
+from .utils import clone_repo
+
+pytestmark = pytest.mark.benchmark
+
+tests = [
+    {
+        "name": "Mentat: replace subprocess/git with GitPython",
+        "codebase_url": "http://github.com/AbanteAI/mentat",
+        "codebase_name": "mentat",
+        "commit": "",  # hexsha
```

Comment:
Is the plan to add the commit this PR ends up being when the PR is merged?

I guess usually if adding a new benchmark you could use the latest commit on Mentat, as long as when adding the benchmark you don't change anything else.

Reply:
Oh, if we follow your comment from above (`repos_for_benchmarks`) this wouldn't be an issue, that'd be great.